pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -151,6 +152,27 @@ func updateVitessShard(key client.ObjectKey, vts *planetscalev2.VitessShard, vtk
 	// Update labels, but ignore existing ones we don't set.
 	update.Labels(&vts.Labels, newShard.Labels)
 
+	// Remove old annotations that shouldn't be there that we injected previously.
```

Comment:
Add another line to document the implicit ordering dependency:

```suggestion
	// Remove old annotations that shouldn't be there that we injected previously.
	// This must be done before we update vts.Spec.
```

Reply:
Added explanation comment about implicit ordering: https://github.com/planetscale/vitess-operator/pull/51/commits/e12e40a8d7d1f5b6127b6a00b1f8bbd4481cb875.